fetch.bs

```diff
@@ -7346,6 +7357,10 @@ constructor steps are:
  <li><p>If <var>init</var>["{{RequestInit/signal}}"] <a for=map>exists</a>, then set
  <var>signal</var> to it.
 
+ <li><p>If <var>init</var>["{{RequestInit/priority}}"] <a for=map>exists</a>, then use
+ it appropriately in setting <var>request</var>'s <a for=request>internal priority</a>
+ to an <a>implementation-defined</a> object.
```

Comment:
But this is a subset of what the fetch algorithm takes into account. Is that what we want? An alternative might be that the fetch algorithm also has a branch when internal priority is an implementation-defined object already and then modifies it to take into account potentially changed signals.

Reply:
I tweaked the logic to only update the existing internal priority if one was already set.  If it is null then the existing fetching logic will take all of the inputs available to construct one.  If an existing one is updated, the additional signals aren't explicitly copied so I left it vague for implementations to decide how much state they need to maintain with the implementation-defined object to be able to modify an existing internal priority with a (possibly) new priority signal.

I think this walks the line without having to require that all of the underlying signals in an existing Request be copied over while still allowing for priority to tweak it.

Reply:
I think that works, but then you do need to set it on the request if internal priority is null, right?

Reply:
Argh, yep.  Think I got it this time. The check for when `init['priority']` is set and using an existing request in the constructor now updates internal priority if it was already set, otherwise it sets the value of priority to be what was specified in init.